### ISSUE TITLE
decompilation error

### ISSUE THREAD

**Issue by antonmr** [skylot/jadx#655]
Hi, I'm trying to decompile an APK but it's giving lots of errors.

![image](https://user-images.githubusercontent.com/12773308/57377298-81404d80-7178-11e9-8c91-9b897915f42b.png)

[Validator.zip](https://github.com/skylot/jadx/files/3157372/Validator.zip)

**Comment by antonmr** [comment]
Yes there are erros. I sent the WARNING example, but it should be this one.
The next print is an example of one of the errors, but there are whole functions (like this one) that are decompiled incorrectly, also has several Anonymous classes and some functions that looks like it's not entirely correct.

![image](https://user-images.githubusercontent.com/12773308/57528947-060ea100-730a-11e9-9949-71d7698834f3.png)

[sc.txt](https://github.com/skylot/jadx/files/3166577/sc.txt)

**Comment by jpstotz** [comment]
I assume you were using jadx 0.9.0 - use the [latest unstable version](https://github.com/skylot/jadx#Downloads) instead. It is able to decompile the method. However I doubt that it is 100% correct because it is too complex. 

Therefore don't try to re-compile the app using Android Studio this will most likely don't work.

**Comment by antonmr** [comment]
Just tried out the latest unstable version. Like the example below, there are lots of infos and warnings, BUT the warnings say that the code was lost, parts were removed etc.
Maybe it's missing something from the code?

![image](https://user-images.githubusercontent.com/12773308/57623475-36eb1200-7566-11e9-9cb5-ddc09a6feb6e.png)

**Comment by jpstotz** [comment]
As I said in my last post:

> However I doubt that it is 100% correct because it is too complex.

Jadx can decompile a large percentage of Android byte code, however it can not decompile each and every method you find in Android apps. Skylot and the other developers work hard to improve the decompilation quality but they can't do magic. Therefore each improvement takes it's time.
